core/debug.cc

```diff
@@ -371,21 +372,37 @@ static bool SkipSymbol(char *symbol) {
   _exit(EXIT_FAILURE);
 }
 
+[[noreturn]] static void abort_failure() {
+  abort();
+}
+
 [[ gnu::noinline, noreturn ]] void GoPanic() {
   if (oops_msg == "")
     oops_msg = DumpStack();
 
-  // Create a crash log file
-  try {
-    std::ofstream fp(P_tmpdir "/bessd_crash.log");
-    fp << oops_msg;
-    fp.close();
-  } catch (...) {
-    // Ignore any errors.
+  // Create a crash log file if we are not dropping a core file.
+  if (!FLAGS_core_dump) {
+    try {
+      std::ofstream fp(P_tmpdir "/bessd_crash.log");
+      fp << oops_msg;
+      fp.close();
+    } catch (...) {
+      // Ignore any errors.
+    }
   }
 
-  // The default failure function of glog calls abort(), which causes SIGABRT
-  google::InstallFailureFunction(exit_failure);
+  // Set SIGABRT back to the default to avoid catching the abort used to
+  // generate the core file.
+  struct sigaction sa;
+  memset(&sa, 0, sizeof(sa));
+  sa.sa_handler = SIG_DFL;
+  sigaction(SIGABRT, &sa, 0);
+
+  if (FLAGS_core_dump) {
+    google::InstallFailureFunction(abort_failure);
```

Comment:
If I remember correctly, this line is somewhat redundant (`LOG(FATAL) << ...` below will invoke `abort()` eventually). But this explicit code looks better.

Reply:
This is what was commented but if I did not call InstallFailureFunction with abort_failure I got some other strange behavior. Given that you like the explicit code anyway I will keep it as it is.